CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLWorker.cc

```diff
@@ -772,13 +771,16 @@ const std::pair<LocalPoint, LocalPoint> SiPixelLorentzAnglePCLWorker::surface_de
 void SiPixelLorentzAnglePCLWorker::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
   edm::ParameterSetDescription desc;
   desc.add<std::string>("folder", "AlCaReco/SiPixelLorentzAngle");
+  desc.add<bool>("notInPCL", false);
   desc.add<std::string>("fileName", "testrun.root");
   desc.add<edm::InputTag>("src", edm::InputTag("TrackRefitter"));
   desc.add<double>("ptMin", 3.);
   desc.add<double>("normChi2Max", 2.);
-  desc.add<int>("clustSizeYMin", 4);
+  desc.add<double>("clustSizeYMin", 3.999999);
+  desc.add<double>("clustSizeYMinL4", 2.999999);
+  desc.add<double>("clustSizeXMax", 5.0);
   desc.add<double>("residualMax", 0.005);
-  desc.add<double>("clustChargeMax", 120000);
+  desc.add<double>("clustChargeMax", 50000);
```

Comment:
This a bit misleading naming (from Morris' code), because this is the clustChargeMax per unit length! So in the end the cluster charge could be bigger see line https://github.com/CMSTrackerDPG/cmssw/pull/16/files/201ada0304e3d4590f6862ee534de7dafe66b314..1771adbf05f9b238251e405e7d8c242dfaf80662#diff-834cd3677d4266740a6bfc6853af3e73236a9c52ddf14d844a64788db2465ab2R493

So can we just call this clustChargeMaxPerLenght?

Reply:
Is it per length or per area because in the line you pointed out I see `* drdz`. In that case it would be the charge area density.

Reply:
This looks more like a scale factor of r/z to me, instead of unit length or unit area.

where r = sqrt (x^2 + y^2 + z^2)

Reply:
I think that was not an area, if you read Morris' comments in his code, he says it somewhere that it's meant to be a charge per unit lenght

Reply:
here are the details `double drdz = sqrt(1.+cotalpha*cotalpha+cotbeta*cotbeta);` see
https://github.com/wweiphy/LA/blob/0613f0c86017179b9fb094b69fd77bccffb5893f/calculateEfieldFromNtpl_template_2018E1_prob.C#L579
and the comment here
`float clusterCharge_cut = 50.;  //charge in ke per unit thickness`
https://github.com/wweiphy/LA/blob/0613f0c86017179b9fb094b69fd77bccffb5893f/calculateEfieldFromNtpl_template_2018E1_prob.C#L246

Reply:
Just want to make sure: is unit thickness the width of one pixel? 
And by multiplying charge per thickness with drdz (which is related to the track angle) we get a better cut in terms of track direction? 

From Morris's reply to my previous questions he said the new charge cut is independent of the track angle, but according to the math it is related. Now I'm getting confused.

Reply:
dr/dz is the cosine of the grazing angle, do Tamas is right, it is cluster charge normalized to the unit path length of traversed silicon. The units are still electrons (it has nothing to do with area), it's just the definition of the quantity that involves normalization.

![image](https://user-images.githubusercontent.com/5082376/127115284-1fccb4e8-1a84-4a1a-830b-26ec267e20d8.png)

Reply:
But from the definition `double drdz = sqrt(1. + cotalpha * cotalpha + cotbeta * cotbeta);` this cannot be a cosine (value >1). What Wei wrote looks correct to me. This is r/z or, in other words, the length of the traversed path in silicon in units of the sensor thickness z. From `double clusterCharge_cut = clustChargeMax_ * drdz;` I conclude that the charge cut value is "normalized" based on the traversed path length. At least for my preference the logic here is a bit twisted because I would prefer dividing the cluster charge by `drdz` and keeping the cut value fixed but fine, the final effect should be the same. The definition of `drdz` is consistent with Fig. 2 from http://cms.cern.ch/iCMS/jsp/openfile.jsp?type=NOTE&year=2007&files=NOTE2007_033.pdf
![image](https://user-images.githubusercontent.com/4885289/127127636-43302d6c-5c9c-4024-9e0a-5007bc108511.png)

Reply:
Surely has nothing to do with area though

Reply:
> This is r/z or, in other words, the length of the traversed path in silicon in units of the sensor thickness

I think we are saying the same thing. I called it dr/dz

Reply:
I agree, I just had a cursory look without diving deeper into it and interpreted `drdz` as `dr * dz` which was clearly a wrong assumption.